public/main.js

```diff
@@ -428,7 +444,7 @@ console.log(message);
   State.inputModel = monaco.editor.createModel(
     UI.getInitialCode(),
     "typescript",
-    monaco.Uri.file("input.tsx"),
+    monaco.Uri.file("input.ts"),
```

Comment:
I think here you should also set the file extension based on the `compilerOptions`, like you did before.

Because otherwise it does not work on init (with JSX option set initially): https://typescript-play-pr-35.surge.sh/?jsx=2#example/jsx.

Reply:
Yep, [done](https://github.com/agentcooper/typescript-play/pull/35/commits/124ce730f5f01ee73f5758536f5eba1638d6b740).

Reply:
Okay [fixed the typo](https://github.com/agentcooper/typescript-play/pull/35/commits/f11510e209573889a7637f19987bb4ac579adb53). The temporary site is live, [works](https://typescript-play-pr-35.surge.sh/?jsx=2#example/jsx)

Reply:
Awesome, thanks. Now maybe just extract that common piece into something like `function createFile(compilerOptions)`?

Reply:
No problems. [Extracted the logic](https://github.com/agentcooper/typescript-play/pull/35/commits/a7bf337bed38a930e96654d18cc0e7e46f66edfe).

Reply:
Thanks! Merged and deployed.

Reply:
Welcome! Happy to help!